src/analyze/analyze-capability.c

```diff
@@ -20,27 +32,49 @@ int verb_capabilities(int argc, char *argv[], void *userdata) {
         /* Determine the maximum of the last cap known by the kernel and by us */
         last_cap = MAX((unsigned) CAP_LAST_CAP, cap_last_cap());
 
-        if (strv_isempty(strv_skip(argv, 1)))
+        if (arg_capability == ARG_CAP_MASK || (argc == 2 && argv[1][0] == '@')) {
+                uint64_t cap_mask;
+
+                if (argc != 2)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Exactly 1 positional argument expected.");
+
+                if (argv[1][0] == '@')
+                        ++argv[1];
+                if (safe_atoux64(argv[1], &cap_mask))
```

Comment:
please check for errors with a `< 0` check. And also pass the return value as first arg to the log_error_errno() below.

i.e.:

```c
r = safe_atoux64(…);
if (r < 0)
        return log_error_errno(r, …
```